packages/plugins/rudder_plugin_interface/lib/src/models/rudder_config.dart

```diff
@@ -119,6 +120,20 @@ class RudderConfig {
       _mobileConfigMap['controlPlaneUrl'] = controlPlaneUrl;
       _webConfigMap['configUrl'] = controlPlaneUrl;
     }
+    if (Utils.isEmpty(dataResidency)) {
+      RudderLogger.logError(
+          "dataResidency can not be null or empty. Set to default.");
+      _mobileConfigMap['dataResidency'] = Constants.DATA_RESIDENCY_VALUE;
+      _webConfigMap['dataResidency'] = Constants.DATA_RESIDENCY_VALUE;
+    } else if (dataResidency != "US" && dataResidency != "EU") {
```

Comment:
US and EU can be constant values

Reply:
yes, this was the minor improvement I was talking about

Reply:
yes, changed them to enums